src/backend/utils/mmgr/mcxt.c

```diff
@@ -801,6 +894,9 @@ MemoryContextStats_recur(MemoryContext topContext, MemoryContext rootContext,
 					 */
 
 					MemoryContext_LogContextStats(siblingCount, cumAllAllocated, cumAllFreed, cumCurAvailable, prevChildName);
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
+					MemoryContext_printTopListOfChunks(prev_chunk_htable);
```

Comment:
Yes, it seems we can move MemoryContext_printTopListOfChunks call to MemoryContext_LogContextStats.
In this line we can do a simple switch, like:
```
chunk_table = prev_chunk_htable;
MemoryContext_LogContextStats(); //which always call MemoryContext_printTopListOfChunks with chunk_table , or we may rewrite MemoryContext_printTopListOfChunks to use global variable, destroy it and NULL it
```